docs/book/src/developer/providers/v1.1-to-v1.2.md

```diff
@@ -55,20 +55,82 @@ in ClusterAPI are kept in sync with the versions used by `sigs.k8s.io/controller
+  requires to dry-run changes on templates, and it is required to implement following changes in order to prevent dry-run
+  to return errors due to immutability checks implemented in the web hooks.
+
+  E.g. in CAPD following changes should be applied to the DockerMachineTemplate webhook:
+
+  ```diff
+  + type DockerMachineTemplateWebhook struct{}
+
+  + func (m *DockerMachineTemplateWebhook) SetupWebhookWithManager(mgr ctrl.Manager) error {
+  - func (m *DockerMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
+        return ctrl.NewWebhookManagedBy(mgr).
+  -         For(m).
+  +         For(&DockerMachineTemplate{}).
+  +         WithValidator(m).
+            Complete()
+  }
+
+    // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-dockermachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockermachinetemplates,versions=v1beta1,name=validation.dockermachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+  + var _ webhook.CustomValidator = &DockerMachineTemplateWebhook{}
+  - var _ webhook.Validator = &DockerMachineTemplate{}
+
+  + func (*DockerMachineTemplateWebhook) ValidateCreate(ctx context.Context, _ runtime.Object) error {
+  - func (m *DockerMachineTemplate) ValidateCreate() error {
+        ...
+    }
+
+  + func (*DockerMachineTemplateWebhook) ValidateUpdate(ctx context.Context, oldRaw runtime.Object, newRaw runtime.Object) error {
+  +     newObj, ok := newRaw.(*DockerMachineTemplate)
+  +     if !ok {
+  +         return apierrors.NewBadRequest(fmt.Sprintf("expected a DockerMachineTemplate but got a %T", newRaw))
+  +     }
+  - func (m *DockerMachineTemplate) ValidateUpdate(oldRaw runtime.Object) error {
+        oldObj, ok := oldRaw.(*DockerMachineTemplate)
+        if !ok {
+            return apierrors.NewBadRequest(fmt.Sprintf("expected a DockerMachineTemplate but got a %T", oldRaw))
+        }
+        ...
+        // Immutability check
+  +     if !topology.ShouldSkipImmutabilityChecks(ctx, newObj) &&
+  +        !reflect.DeepEqual(newObj.Spec.Template.Spec, oldObj.Spec.Template.Spec) {
+  -     if !reflect.DeepEqual(m.Spec.Template.Spec, old.Spec.Template.Spec) {
+            allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec"), m, dockerMachineTemplateImmutableMsg))
+        }
+        ...
+    }
+
+  + func (*DockerMachineTemplateWebhook) ValidateDelete(ctx context.Context, _ runtime.Object) error {
+  - func (m *DockerMachineTemplate) ValidateDelete() error {
+        ...
+    }
+  ```
+
+NOTES:
+
+- We are introducing a `DockerMachineTemplateWebhook` struct because we are going to use a controller runtime
+  `CustomValidator` that will allow to skip the immutability check only when the topology controller is dry running
+  while preserving the validation behaviour for all the other cases.
```

Comment:
```suggestion
- We are introducing a `DockerMachineTemplateWebhook` struct because we are going to use a controller runtime
  `CustomValidator`. This will allow to skip the immutability check only when the topology controller is dry running
  while preserving the validation behaviour for all other cases.
```
nit